.readthedocs.yaml

```diff
@@ -1,16 +1,31 @@
-version: 2
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
 
-python:
-  install:
-    - method: pip
-      path: .
-      extra_requirements:
-        - docs
+# Required
+version: 2
 
+# Set the version of Python and other tools you might need
 build:
   os: ubuntu-22.04
   tools:
     python: "3.11"
+  jobs:
+    pre_install:
+      - python3 .sphinx/build_requirements.py
+      - git fetch --unshallow || true
 
+# Build documentation in the docs/ directory with Sphinx
 sphinx:
-  configuration: docs/conf.py
+  builder: dirhtml
+  configuration: conf.py
+  fail_on_warning: true
+
+# If using Sphinx, optionally build your docs in additional formats such as PDF
+formats:
+   - pdf
+
+# Optionally declare the Python requirements required to build your docs
```

Comment:
can we keep those requirements in the pyproject.toml under a docs section please? instead of spreading requirements into different files

Reply:
would you rather i do that in this PR or handle it in a separate one? i'll have to write a parser to use in custom_conf.py and it seems like it would be harder to review in a single PR